src/chainstate/burn/db/sortdb.rs

```diff
@@ -3794,14 +3794,44 @@ impl SortitionDB {
         burnchain: &Burnchain,
         block: &BlockSnapshot,
     ) -> Result<bool, db_error> {
-        let reward_start_height = burnchain.reward_cycle_to_block_height(
+        let mut reward_start_height = burnchain.reward_cycle_to_block_height(
```

Comment:
this can be re-written to not use a mutable variable... e.g., call this `prospective_reward_start_height`